File: 2nd year/1st semester/MAP/ToyLanguageInterpreter/src/View/RunExample.java
package View;

import Controller.InterpreterController;

public class RunExample extends Command {
    private InterpreterController ctrl;

    RunExample(String key, String desc, InterpreterController ctrl){
        super(key, desc);
        this.ctrl = ctrl;
    }

    @Override
    public void execute() throws java.lang.InterruptedException {
        /*
        try {
            System.out.println(this.ctrl.getRepo().getCurrentPs());
            ctrl.executeAll();
            System.out.println(this.ctrl.getRepo().getCurrentPs());
        }
        catch (Exception e) {
            System.out.println(e.toString());
        }
        */

        //this.ctrl.getRepo().getProgramStates().forEach(prg -> System.out.println(prg.toString()));
        this.ctrl.executeAllSteps();
        //this.ctrl.getRepo().getProgramStates().forEach(prg -> System.out.println(prg.toString()));
    }
}

File: 2nd year/1st semester/MAP/CS_ToyLanguageInterpreter/MAP_Interpreter/View/RunExample.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MAP_Interpreter.View
{
    public class RunExample : Command
    {
        private InterpreterController ctrl;

        public RunExample(string key, string description, InterpreterController ctrl) : base(key, description)
        {
            this.ctrl = ctrl;
        }

        public override void execute()
        {
            try
            {
                Console.WriteLine(ctrl.getRepository().getProgramState().ToString());
                ctrl.executeAll();
                Console.WriteLine(ctrl.getRepository().getProgramState().ToString());
            }
            catch (Exception e)
            {
                Console.WriteLine(e.ToString());
            }
        }
    }
}
